modules/geoipbackend/geoipbackend.cc

```diff
@@ -261,24 +261,30 @@ void GeoIPBackend::initialize() {
 
     // finally fix weights
     for(auto &item: dom.records) {
-      float weight=0;
-      float sum=0;
+      map<uint16_t, float> weights;
+      map<uint16_t, float> sums;
+      map<uint16_t, GeoIPDNSResourceRecord> lasts;
       bool has_weight=false;
       // first we look for used weight
       for(const auto &rr: item.second) {
-        weight+=rr.weight;
+        weights[rr.qtype.getCode()] += rr.weight;
         if (rr.has_weight) has_weight = true;
       }
       if (has_weight) {
         // put them back as probabilities and values..
         for(auto &rr: item.second) {
-          rr.weight=static_cast<int>((static_cast<float>(rr.weight) / weight)*1000.0);
-          sum += rr.weight;
+          uint16_t rr_type = rr.qtype.getCode();
+          rr.weight=static_cast<int>((static_cast<float>(rr.weight) / weights[rr_type])*1000.0);
```

Comment:
This was already the case with the existing code, but it looks like we would happily divide by `0.0` if the weight of all records for a given type is set to `0`.

Reply:
What would be the desired behavior in that case?
Maybe the yaml parser should just ignore records with a weight of 0?


Anyway it seems out of the scope of this review, what do you think @rgacogne?

Reply:
I think the parser should error out on a weight of `0`.

I agree it's a bit out of the scope so if you don't feel like fixing that in your PR just let me know and I'll either open a new PR or an issue so we don't forget to fix it.

Reply:
OK I'm doing it in another review :)